qiskit/algorithms/evolvers/variational/var_qite.py

```diff
@@ -0,0 +1,120 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Variational Quantum Imaginary Time Evolution algorithm."""
+from functools import partial
+from typing import Optional, Callable
+
+import numpy as np
+from scipy.integrate import OdeSolver, RK45
+
+from qiskit.opflow import (
+    StateFn,
+    ExpectationBase,
+)
+from qiskit.algorithms.aux_ops_evaluator import eval_observables
+from qiskit.algorithms.evolvers import EvolutionProblem, EvolutionResult
+from qiskit.algorithms.evolvers.imaginary_evolver import ImaginaryEvolver
+from qiskit.utils import QuantumInstance
+from .solvers.ode.ode_function_factory import OdeFunctionFactory
+from .var_qte import VarQTE
+from .variational_principles.imaginary_variational_principle import (
+    ImaginaryVariationalPrinciple,
+)
+
+
+class VarQITE(ImaginaryEvolver, VarQTE):
+    """Variational Quantum Imaginary Time Evolution algorithm."""
+
+    def __init__(
+        self,
+        variational_principle: ImaginaryVariationalPrinciple,
+        ode_function_factory: OdeFunctionFactory,
+        ode_solver_callable: OdeSolver = RK45,
```

Comment:
Since we use SciPy method eventually, I followed their types partly:
 ```
solve_ivp(fun: Callable,
              t_span: Any,
              y0: Union[ndarray, Iterable, int, float],
              method: Union[str, OdeSolver, None] = 'RK45',
              ...
```
They use it as follows:
`solver = method(fun, t0, y0, tf, vectorized=vectorized, **options)`.

The suggested `RK45()` does not work bc of required arguments.
I could extend our type hint here to `str`.

Reply:
Adding strings works for me, I just thought that the previous default value (`RK45`) was a `type` and not a `OdeSolver` 🤔